### DISCUSSION TITLE
How to deal with linebreaks when convertig a marked text to a block elmenet (headline)

### DISCUSSION THREAD

**Discussion by lanwin** [ianstormtaylor/slate#5761] [Q&A] [answered]
Hi, I have the following case and can not get a proper solution. Maybe someone can get me a hint.

Initial my editor has the following state.

`[{type:"p",children:[{text:"test1\ntest2\ntest3"}]]`

Now the users select `test2` and chooses make headline witch basically executes this transform:

`Transforms.setNodes(editor, { type: "h4" }, { mode: 'lowest', voids: false, split: true });`

Now the doc looks like this:

```
[ 
  {type:"p",children:[{text:"test1\n"}]
  {type:"h4",children:[{text:"test2"}]
  {type:"p",children:[{text:"\ntest3"}]
]
```

My problem is that I now have a empty line above and below the h4. This is cause the \n arent getting removed.

Whats the best why to ensure that the newlines getting handled correctly?

Thank you!

**Comment by WindRunnerMax** [comment]
You shouldn't use any \n markers; the initial value should be

```js
[ 
  {type:"p",children:[{text:"test1"}],
  {type:"p",children:[{text:"test2"}],
  {type:"p",children:[{text:"test3"}]
]
```

**Reply by lanwin** [reply]
Yes in an ideal world that would be true. But this does not answer my question.

**Reply by gerardnico** [reply]
It does. Changing your initial json is the solution (you don't implement formatting in your data but with your nodes)

Otherwise, you can always implement your own logic (ie you change the text before the transform and you wrap all that in your command.

**Comment by lanwin** [comment, marked as answer]
For anyone else having this problem, the following code did it.

```typescript
Transforms.setNodes(editor, { type: "h4" }, { mode: 'lowest', voids: false, split: true });

const after = Editor.after(editor, editor.selection);
if (after) {
  const [nodeAfter, pathAfter] = Editor.node(editor, after);
  if (Text.isText(nodeAfter) && nodeAfter.text.startsWith('\n'))
	Transforms.delete(editor, { at: { path: pathAfter, offset: 0 }, distance: 1 });
}
const before = Editor.before(editor, editor.selection);
if (before) {
  const [nodeBefore, pathBefore] = Editor.node(editor, before);
  if (Text.isText(nodeBefore) && nodeBefore.text.endsWith('\n'))
	Transforms.delete(editor, { at: { path: pathBefore, offset: nodeBefore.text.length - 1 }, distance: 1 });
}
```
